tests/population/test_base_population.py

```diff
@@ -378,3 +378,26 @@ def _check_locations(simulants):
             1 / len(simulants.location.unique()),
             abs_tol=0.01,
         )
+
+
+@pytest.mark.parametrize("age_start, age_end", [(0, 100), (2, 52), (21, 67), (12, 48)])
```

Comment:
I pulled this branch to try and understand the new logic in data_tranformations.py and found that this test passes even when I commented that new logic out. i.e. we need a test that fails prior to the logic but passes with it